site/app/views/PollView.php

```diff
@@ -0,0 +1,81 @@
+        return $this->core->getOutput()->renderTwigTemplate("polls/AllPollsPageInstructor.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/polls',
+            'todays_polls' => $todays_polls,
+            'older_polls' => $older_polls,
+            'future_polls' => $future_polls
+          ]);
+    }
+
+    public function showPollsStudent($todays_polls, $older_polls) {
+        $this->core->getOutput()->addBreadcrumb("Polls");
+        $this->core->getOutput()->addInternalCss('polls.css');
+        $this->core->getOutput()->addInternalJs('polls-dropdown.js');
+        $this->core->getOutput()->enableMobileViewport();
+        return $this->core->getOutput()->renderTwigTemplate("polls/AllPollsPageStudent.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/polls',
+            'todays_polls' => $todays_polls,
+            'older_polls' => $older_polls,
+            'user_id' => $this->core->getUser()->getId(),
+          ]);
+    }
+
+    public function showNewPollPage() {
+        $this->core->getOutput()->addBreadcrumb("Polls", $this->core->buildCourseUrl(["polls"]));
+        $this->core->getOutput()->addBreadcrumb("New Poll");
+        $this->core->getOutput()->addInternalCss('polls.css');
+        return $this->core->getOutput()->renderTwigTemplate("polls/NewPollPage.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/polls',
+          ]);
+    }
+
+    public function showPollStudent($poll) {
+        $this->core->getOutput()->addBreadcrumb("Polls", $this->core->buildCourseUrl(["polls"]));
+        $this->core->getOutput()->addBreadcrumb("View Poll");
+        return $this->core->getOutput()->renderTwigTemplate("polls/PollPageStudent.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/polls',
+            'poll' => $poll,
+            'user_id' => $this->core->getUser()->getId(),
+          ]);
+    }
+
+    public function editPoll($poll) {
+        $this->core->getOutput()->addBreadcrumb("Polls", $this->core->buildCourseUrl(["polls"]));
+        $this->core->getOutput()->addBreadcrumb("Edit Poll");
+        $this->core->getOutput()->addInternalCss('polls.css');
+        return $this->core->getOutput()->renderTwigTemplate("polls/NewPollPage.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/polls',
+            'poll' => $poll
+          ]);
+    }
+
+    public function viewResults($poll, $results) {
+        $this->core->getOutput()->addBreadcrumb("Polls", $this->core->buildCourseUrl(["polls"]));
+        $this->core->getOutput()->addBreadcrumb("View Results");
+        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('plotly', 'plotly.js'));
```

Comment:
It would be nice if you added `$this->core->getOutput()->enableMobileViewport();` I am not sure how well plotly works with with mobile though so it might not work